### ISSUE TITLE
Allow multiple instances of Pianoteq - or in fact all instruments

### ISSUE THREAD

**Issue by IFW-SPS** [zynthian/zynthian-issue-tracking#1233]
Currently only a single instance of Pianoteq instrument is allowed and once one is added, the option to add Pianoteq is no longer available.

Please don't limit instruments/engines artificially, unless they'd e.g. crash. 

A few use cases:
1, split keyboard, play piano live, plus play/trigger guitar chords
2, sequence guitar, harp etc and play piano live.
There is a high chance that most of the time the piano would play, and once every few bars guitar chords or anything else (like in typical sheet music) - so it’s not like all the instances are playing all the time. Although I have no idea how Pianoteq internally works, but presumably when Pianoteq is not playing and all the sounds rang out it is not taking many resources.

**Comment by wanthalf** [comment]
I agree that limiting options just because of limited resources (performance, memory) is not a "solution" to anything. First, there are engines and patches installed on Zynthian that do not play well even when run completely alone and exclusively on the newest Pi5 with full overclocking - and still nobody removes them. Second, the performance and resources may always change - at least in theory (right, the Raspberry Pi does not get "upgrades" very often and cannot be really "upgraded" in custom ways, nor really significantly overclocked). The load of different engines may vary independently of their amount - a single engine may exhaust the resources more than several other engines running in parallel.

**Comment by riban-bw** [comment]
Single instance is an historical constraint that was used to limit instances for various reasons, not just resource usage. It is used for several engines and there is a technical reason for each. For example, Pianoteq has a JSON-RPC interface that zynthian must manage. This was introduced in a fairly recent version of Pianoteq. To get this working, we implemented a fairly simple interface, using a static address and port. Each connection needs to be managed and this will need to be added to the code. We now have a fairly stable integration of Pianoteq so we can consider implementing the extra code required to remove this constraint. There is usually a reason for everything which may sometimes be less obvious than first sight.

**Comment by pft** [comment]
Just discovered pianoteq (pro and standard(?)) has layering and morphing. Meaning with just one instance you can get multitimbrality within pianoteq.

I do not know whether all controls could be made available via the zynthian interface, such as a crossfade control, but at least you can make and save a preset in VNC, which is then available from Zynthian.

Does not help with splits through as far as I can see.
